pyproject.toml

```diff
@@ -36,7 +36,7 @@ dev = [
     "nbstripout",
     "pytest",
     "pytest-cov",
-    "ruff",
+    "ruff==0.0.284",
```

Comment:
You updated the pre-commit config but left this pinned. So we still have the issue of incompatible pre-commit and project. My suggestion is to be more explicit in our project so the alignment is guaranteed then figure out an easier way to keep them aligned in the future as a follow up

Reply:
I think I have it. I pinned ruff but pre-commit autoupdate gave a conflicting version. So, `pre-commit autoupdate --repo https://github.com/psf/black` to update black only should do the trick.

Reply:
The tests pass now but we still have the possibility of this pre-commit/black mismatch occurring again. Can you pin the version of black in this file to match the one in the pre-commit. We can figure out if there is an automated way to align these as follow up. In my opinion it would be nicer to keep the version of black that was originally in in the pre-commit and just pin it here then roll back the changes from the newer version of black. That way the content changes and formatting churned aren't coupled together.